managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -2661,6 +2708,34 @@ void setReadPosition(Position newReadPositionInt) {
         }
     }
 
+    void clearIncompleteAckedRecordsByLedgerId(final long ledgerId){
+        lock.writeLock().lock();
+        try {
+            List<Range> rangeListToDelete = individualDeletedMessages.asRanges().stream()
+                    .filter(range -> range.lowerEndpoint().getLedgerId() == ledgerId)
+                    .map(range -> Range.openClosed(
+                            new LongPairRangeSet.LongPair(ledgerId, range.lowerEndpoint().getEntryId()),
+                            new LongPairRangeSet.LongPair(ledgerId, range.upperEndpoint().getEntryId())
+                    )).collect(Collectors.toList());
+            if (!rangeListToDelete.isEmpty()) {
+                rangeListToDelete.forEach(individualDeletedMessages::remove);
+            }
+
+            if (batchDeletedIndexes != null) {
+                Set<PositionImpl> batchedIndexesToDelete = batchDeletedIndexes.keySet().stream()
+                        .filter(position -> position.getLedgerId() == ledgerId)
+                        .collect(Collectors.toSet());
+                if (!batchedIndexesToDelete.isEmpty()) {
+                    batchedIndexesToDelete.forEach(batchDeletedIndexes::remove);
+                }
+            }
+        } catch (Exception e){
+            e.printStackTrace();
```

Comment:
why `e.printStackTrace();`?

Reply:
This was added during debugging, and I forgot to delete it， already fixed